skglm/penalties/separable.py

```diff
@@ -229,6 +228,84 @@ def alpha_max(self, gradient0):
         return np.max(np.abs(gradient0))
 
 
+spec_SCAD = [
+    ('alpha', float64),
+    ('gamma', float64)
+]
+
+
+@jitclass(spec_SCAD)
+class SCAD(BasePenalty):
+    """Smoothly Clipped Absolute Deviation.
+
+    Notes
+    -----
+    With x >= 0
+    pen(x) =
+    alpha * x                         if x =< alpha
+    (2 * gamma * alpha * x - x ** 2 - alpha ** 2) \
+        / (2 * (gamma - 1))           if alpha < x < alpha * gamma
+    (alpha ** 2 * (gamma + 1)) / 2    if x > gamma * alpha
+    value = sum_{j=1}^{n_features} pen(abs(w_j))
+    """
+
+    def __init__(self, alpha, gamma):
+        self.alpha = alpha
+        self.gamma = gamma
+
+    def value(self, w):
+        """Compute the value of the SCAD penalty at w."""
+        value = np.full_like(w, (self.alpha ** 2 * (self.gamma + 1)) / 2)
+        for j in range(len(w)):
+            if np.abs(w[j]) <= self.alpha:
+                value[j] = self.alpha * np.abs(w[j])
+            elif np.abs(w[j]) > self.alpha and np.abs(w[j]) < self.alpha * self.gamma:
+                value[j] = (
+                    2 * self.gamma * self.alpha * np.abs(w[j])
+                    - np.abs(w[j]) ** 2 - self.alpha ** 2) / (2 * (self.gamma - 1))
+        return np.sum(value)
+
+    def prox_1d(self, value, stepsize, j):
+        """Compute the proximal operator of SCAD penalty."""
+        tau = self.alpha * stepsize
+        g = self.gamma / stepsize
+        if np.abs(value) <= 2 * tau:
+            return ST(value, tau)
+        if np.abs(value) > g * tau:
+            return value
+        return ((g - 1) / (g - 2)) * ST(value, (g * tau) / (g - 1))
```

Comment:
sure ? 
In wen's review I see 
![image](https://user-images.githubusercontent.com/8993218/169352994-24250e2c-a022-4f3c-a825-76415d394e79.png)

(a is gamma) 
I'm not sure this matches what we have written

Reply:
calculation with stepsize is given here https://arxiv.org/pdf/1303.4434.pdf:
I do not think there is an easier way.